From 845d91e04a2c4033f5c73d52d85b2cd73fffac70 Mon Sep 17 00:00:00 2001
From: OpenWrt community <openwrt-devel@lists.openwrt.org>
Date: Thu, 16 Mar 2023 16:21:50 +0530
Subject: [PATCH 039/500] OpenWrt: 484-mtd-spi-nor-add-esmt-f25l16pa.patch

spi-nor/esmt.c: add esmt f25l16pa

This fixes support for Dongwon T&I DW02-412H which uses F25L16PA(2S)
flash.

Change-Id: Ie0f58aafd2f000133413280080733ed0f86b3c99
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 629f76c864cf22547955550a6673d2b70e8e795a)
---
 drivers/mtd/spi-nor/esmt.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/esmt.c b/drivers/mtd/spi-nor/esmt.c
index fcc3b0e7cda9..67d9b7eeb853 100644
--- a/drivers/mtd/spi-nor/esmt.c
+++ b/drivers/mtd/spi-nor/esmt.c
@@ -10,6 +10,9 @@
 
 static const struct flash_info esmt_nor_parts[] = {
 	/* ESMT */
+	{ "f25l16pa-2s", INFO(0x8c2115, 0, 64 * 1024, 32)
+			FLAGS(SPI_NOR_HAS_LOCK)
+			NO_SFDP_FLAGS(SECT_4K) },
 	{ "f25l32pa", INFO(0x8c2016, 0, 64 * 1024, 64)
 		FLAGS(SPI_NOR_HAS_LOCK | SPI_NOR_SWP_IS_VOLATILE)
 		NO_SFDP_FLAGS(SECT_4K) },
-- 
2.34.1

